web/src/app/admin/admin-message-logs/AdminMessageLogsGraph.tsx

```diff
@@ -77,24 +81,23 @@ export default function AdminMessageLogsGraph(): JSX.Element {
       statsInput: {
         bucketDuration: graphInterval,
         bucketOrigin: start,
+        segmentBy: segmentBy || null,
       },
     },
   })
   const stats: Stats = data?.messageLogs?.stats?.timeSeries ?? []
 
-  const graphData = React.useMemo(
-    (): MessageLogGraphData[] =>
-      stats.map(({ start, end, count }) => ({
-        count,
-        date: start,
-        label: (
-          <React.Fragment>
-            <Time time={start} /> - <Time time={end} />
-          </React.Fragment>
-        ),
-      })),
-    [stats],
-  )
+  // get list of segment labels from data to map out Lines
+  type LabelDict = { [key: string]: Stats }
+  const [segmentLabels, setSegmentLabels] = useState<LabelDict>({})
+  useEffect(() => {
+    const sl: LabelDict = {}
+    stats.forEach((stat) => {
+      if (!sl[stat.segmentLabel]) sl[stat.segmentLabel] = [stat]
+      else sl[stat.segmentLabel].push(stat)
+    })
+    setSegmentLabels(sl)
```

Comment:
Is there a reason `useMemo` doesn't work here? It should be equal to `useEffect` mixed with `useState` but simpler (since that's it's whole purpose)
https://react.dev/reference/react/useMemo

The main difference is `useMemo` won't cause a double-first-render since it can get it's value inline.

Reply:
You're right